Source/Core/Core/IOS/USB/Emulated/Skylanders/Skylander.cpp

```diff
@@ -1334,15 +1243,17 @@ bool SkylanderPortal::RemoveSkylander(u8 sky_num)
   return false;
 }
 
-u8 SkylanderPortal::LoadSkylander(u8* buf, File::IOFile in_file)
+u8 SkylanderPortal::LoadSkylander(SkylanderFigure* figure)
 {
   std::lock_guard lock(sky_mutex);
 
   u32 sky_serial = 0;
+  u8* block = new u8[0x10];
```

Comment:
You shouldn't new this data because this memory size will never change and creating memory or the heap is more expensive (and in this case you are leaking memory by not deleting it).  Instead I'd suggest:

```
std::array<u8, 0x10> block;
figure->GetBlock(0, block.data());
```

Reply:
Did make it an array, but just a normalc-style array